src/module-elasticsuite-catalog-optimizer/Model/Optimizer/Limitation/Identities.php

```diff
@@ -153,7 +153,11 @@ private function getCategoryIdentities()
 
         $wasAppliedToAllCategories = empty($origData['catalog_view_container']['category_ids']);
 
-        if ($isAppliedToAllCategories || $wasAppliedToAllCategories) {
+        // If the optimizer is or was previously assigned to all but some categories.
+        $appliedToAllCategoriesButSome = ((int) ($data['catalog_view_container']['apply_to'] ?? 0) === 2)
+            || ((int) ($origData['catalog_view_container']['apply_to'] ?? 0) === 2);
+
+        if ($isAppliedToAllCategories || $wasAppliedToAllCategories || $appliedToAllCategoriesButSome) {
```

Comment:
@romainruaud That's the logical thing to do BUT there is something fishy in 2.4.5 (and maybe above) + Luma where the cat_c tag is actually **not** part of the (FPC) cache tags for a category page.
So in effect, cache does not seem to be really purged.